renamed_num_enum/src/lib.rs

```diff
@@ -2,39 +2,54 @@
 
 extern crate alloc;
 
-use ::alloc::format;
-use ::core::convert::TryFrom;
+/// Un-rename `num_enum` when in a no-std crate, since it is not supported yet.
+#[cfg(not(feature = "std"))]
```

Comment:
It seems strange to make this conditional, given `submodule` contains `use ::renamed` which will presumably error if !`std`? I'd be tempted to just remove this `cfg`, if the thing is going to fail anyway...

Reply:
I admit it _is confusing_, but it does allow us to feature an extra test thanks to just these two lines, so I'd say it is worth it (despite confusing code being far from ideal):

  - Since `edition = "2018"`, Cargo will automagically inject `extern crate <crate_name>` for each crate featured in the `Cargo.toml` dependencies. So imagine we have `extern crate renamed;`.

  - Now, when the `std` feature is missing, we (also) emit that line:  `extern crate renamed as num_enum;`.

  - This results, when the `std` feature is missing, in having two names referring to the same external crate 😄

    This allows us to use `::renamed::TryFromPrimitive` _& co._ to refer to our macros, while also making the emitted `::num_enum`-prefixed paths refer to the right thing 👌

Reply:
I still don't entirely see the value, but I also don't think it does any harm, so... 🤷 shipit :)